File: NFCScan/ViewController.m
//
//  ViewController.m
//  NFCScan
//
//  Created by Riffle on 2021/11/18.
//

#import "ViewController.h"
#import "NFCManager.h"

@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    UIButton * scanBtn = [[UIButton alloc] initWithFrame:CGRectMake(100, 200, 100, 30)];
    [scanBtn addTarget:self action:@selector(startScan) forControlEvents:UIControlEventTouchUpInside];
    
    [scanBtn setTitle:@"开始扫描" forState:UIControlStateNormal];
    [scanBtn setTitleColor:[UIColor blackColor] forState:UIControlStateNormal];
    scanBtn.layer.borderColor = [UIColor blackColor].CGColor;
    [self.view addSubview:scanBtn];
}

// 开始扫描
- (void)startScan
{

    [[NFCManager sharedInstance] scanTagWithSuccessBlock:^(NFCNDEFMessage * _Nonnull message) {

        } andErrorBlock:^(NSError * _Nonnull error) {

        }];
}


@end

File: SwiftNFCScan/SwiftNFCScan/ViewController.swift
//
//  ViewController.swift
//  SwiftNFCScan
//
//  Created by liuyalu on 2021/11/18.
//

import UIKit
import CoreNFC

class ViewController: UIViewController {

    var nfcManager:NFCManager?
    override func viewDidLoad() {
        super.viewDidLoad()
        self.nfcManager = NFCManager.init()
        let scanBtn:UIButton = UIButton.init(type: UIButton.ButtonType.custom)
        scanBtn.frame = CGRect(x: 100, y: 60, width: 120, height: 30)
        scanBtn.setTitle("扫描NFC", for: UIControl.State.normal)
        scanBtn.setTitleColor(UIColor.red, for: UIControl.State.normal)
        scanBtn.layer.cornerRadius = 3
        scanBtn.layer.borderWidth = 1
        scanBtn.layer.borderColor = UIColor.blue.cgColor
        self.view.addSubview(scanBtn)
        scanBtn.addTarget(self, action:#selector(ViewController.scanBtnClick), for: UIControl.Event.touchUpInside)
    }
    @objc func scanBtnClick() {
        self.nfcManager?.startScanNFC(successBlock: { ndefMessage in
            
        }, errorBlock: { error in
            
        })
         
     }


}

